### DISCUSSION TITLE
deepin-terminal 特性讨论

### DISCUSSION THREAD

**Discussion by hualet** [linuxdeepin/developer-center#10616] [Features Request & Ideas | 特性请求 & 头脑风暴]
AI

- [ ] in place形式的命令prompt和命令生成
- [ ] 智能命令补全？（要不要默认）
- [ ] 通过prompt生成代码段去执行（有点类似 open-interpretor， 更像是 AI 助手应该干的？不过生成代码还是更适合能看懂的人用--会使用终端的人大概率能看懂）
- [ ] 命令执行出错后生成解释（要不要默认）
- [ ] 针对场景做AI说明和建议，比如 top 场景

命令补全

- [ ] tldr 命令行快速补全？

远程会话

- [ ] ssh 的session 显示水印，方便提醒这不是本机、防止误操作

**Comment by meiyixiang** [comment]
欢迎大家投票，我们会优先排期实现大家迫切需要的功能。

**Comment by ArchieMeng** [comment]
其实这些放到单独的命令行插件去做会更方便些。包括命令行补全也是，bash优化下补全配置或者换fish和zsh(但可能部分终端功能不兼容。记不清是哪些了。)

**Reply by BLumia** [reply]
虽然略off-topic，deepin下默认配置的终端是不是可以换成zsh并提供一些易用的默认配置？

**Reply by hualet** [reply]
> 虽然略off-topic，deepin下默认配置的终端是不是可以换成zsh并提供一些易用的默认配置？

会不会有兼容性的问题……

**Comment by pmzgit** [comment]
能不能把trzsz的兼容性，提升提升，在tmux+trzsz下的远程文件传输，比你们现有方案，效率，易用性高很多
之前适配的issue：
https://github.com/trzsz/trzsz-go/issues/26
项目地址，现在应该很成熟了，官方可以考虑一下：
https://github.com/trzsz/trzsz-go

**Reply by ArchieMeng** [reply]
确实可以看看，谢了。

**Comment by tensor-tech** [comment]
AI扩展终端必要性不大，大部分需求都和AI助手重叠。
